velox/connectors/hive/HiveConnector.h

```diff
@@ -137,6 +137,16 @@ class HivePartitionFunctionSpec : public core::PartitionFunctionSpec {
         channels_(std::move(channels)),
         constValues_(std::move(constValues)) {}
 
+  HivePartitionFunctionSpec(
```

Comment:
It would be nice to add some comments to explain when we use a constructor with 'bucketToPartition' and when without and how the logic is different.